test/index.test.js

```diff
@@ -107,7 +107,80 @@ test('arg truthy correctly parsed', function(t) {
   t.end();
 });
 
-test('env var substition throws on missing env vars', function(t) {
+test('arg same keys', function(t) {
+  resetArgv();
+  process.argv.push('--afoo=new-value');
+
+  let config = loadConfig(__dirname + '/fixture/one');
+  t.equal(config.afoo, 'new-value', 'last arg wins');
+
+  t.end();
+});
+
+test('args unknown', function(t) {
+  resetArgv();
+
+  let config = loadConfig(__dirname + '/fixture/one');
+  t.equal(config.asome, undefined, 'not existing arg is undefined');
+
+  t.end();
+});
+
+test('args whitespaces in assignment', function(t) {
+  resetArgv();
+  process.argv.push('--abar   new-value');
+
+  let config = loadConfig(__dirname + '/fixture/one');
+  t.equal(config.abar, undefined, 'should be undefined');
+
+  t.end();
+});
+
+test('args numbers assignment', function(t) {
+  resetArgv();
+  process.argv.push('--abar 1');
```

Comment:
this is not how the args would be set when executing from the command line. Should be:
```suggestion
  process.argv.push('--abar', '1');
```